### PR TITLE
Test: Fix some Kafka policy issues

### PR DESCRIPTION
**Pull request by eloycoto**
- Make a egress policy from kafka to zook. 
- Added a allow policy from host to kafka (For the healthchecks)

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5156)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: test/k8sT/manifests/kafka-sw-security-policy.yaml [hunk 1]
```diff
@@ -10,7 +10,18 @@ specs:
...
+          "reserved:host": ""
+      toPorts:
+      - ports:
+        - port: "9092"
+          protocol: TCP
+  - endpointSelector:
+      matchLabels:
+        app: kafka
+    egress:
+    - fromEndpoints:
```
Comment (ianvernon) [inline test/k8sT/manifests/kafka-sw-security-policy.yaml]: I know this is WIP, but this will fail - needs to be `toEndpoints` instead of `fromEndpoints`.

Comment (eloycoto) [reply test/k8sT/manifests/kafka-sw-security-policy.yaml]: ouch, thanks

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Github issue
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5551/

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Git issue too: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5554/console

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
hit two times https://github.com/cilium/cilium/issues/5164

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/k8sT/manifests/kafka-sw-security-policy.yaml [hunk 2]
```diff
@@ -7,10 +7,10 @@ specs:
   - endpointSelector:
       matchLabels:
         app: kafka
-    ingress:
```
Comment (ianvernon) [inline test/k8sT/manifests/kafka-sw-security-policy.yaml]: Do we not have to allow ingress from anything with `app: kafka`?

File: test/k8sT/manifests/kafka-sw-security-policy.yaml [hunk 3]
```diff
@@ -4,6 +4,17 @@ description: "Allow only permitted Kafka requests to empire Kafka broker"
 metadata:
   name: "kafka-sw-security-policy"
 specs:
+  - endpointSelector:
```
Comment (ianvernon) [inline test/k8sT/manifests/kafka-sw-security-policy.yaml]: When we run Cilium alongside Kubernetes, we automatically allow ingress from the host to allow for healthchecks / liveness probes to succeed. This is most likely not necessary, but I don't see an issue with explicitly allowing access from the host at the specific port. Have you seen policy denying ingress from the host in the Kafka CI tests? If so, please provide examples where this is the case.

**Comment by ianvernon** [comment]
Can you provide an example of the issues you are fixing? Are there corresponding build failures / tickets that show the nature of the failures? I think it is important to provide a concrete reference (if possible) to the issue that any PR purportedly fixes.

**Comment by eloycoto** [comment]
@ianvernon 

The main reason to allow host is because one day I saw a CI fail that livenessProbe failed, that is why I openned https://github.com/cilium/cilium/issues/4878 and I made this PR to make sure that the livenessProbe works correctly + fix the egress rule.

**Comment by ianvernon** [comment]
@eloycoto OK, but in the future, please link to an issue which documents the failure so reviewers have more context. 

The liveness probe should work correctly regardless, because we always allow ingress from the host when running alongside Kubernetes, so I'm not sure if this will fix much for you. There is nothing technically wrong with explicitly allowing it via policy, though.

**Review by ianvernon — APPROVED** [review]

**Pull request merged**